contracts/introspection/SupportsInterfaceWithLookup.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity ^0.4.23;
+
+import "./ERC165.sol";
+
+
+/**
+ * @title SupportsInterfaceWithLookup
+ * @author Matt Condon (@shrugs)
+ * @dev Implements ERC165 using a lookup table.
+ */
+contract SupportsInterfaceWithLookup is ERC165 {
+  bytes4 public constant InterfaceId_ERC165 = 0x01ffc9a7;
+  /**
+   * 0x01ffc9a7 ===
+   *   bytes4(keccak256('supportsInterface(bytes4)'))
+   */
+
+  /**
+   * @dev a mapping of interface id to whether or not it's supported
+   */
+  mapping(bytes4 => bool) internal supportedInterfaces;
+
+  /**
+   * @dev A contract implementing SupportsInterfaceWithLookup
+   * @dev  implement ERC165 itself
+   */
+  constructor()
+    public
+  {
+    _registerInterface(InterfaceId_ERC165);
+  }
+
+  /**
+   * @dev implement supportsInterface(bytes4) using a lookup table
+   */
+  function supportsInterface(bytes4 _interfaceId)
+    external
+    view
+    returns (bool)
+  {
+    return supportedInterfaces[_interfaceId];
+  }
+
+  /**
+   * @dev private method for registering an interface
+   */
+  function _registerInterface(bytes4 _interfaceId)
```

Comment:
Hate to be a blocker for this but I think this function should `require(_interfaceId != 0xffffffff)` due to the spec of `supportsInterface` ([link](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-165.md#how-a-contract-will-publish-the-interfaces-it-implements)):

```
@return `true` if the contract implements `interfaceID` and
 `interfaceID` is not 0xffffffff, `false` otherwise
```

(Alternatively we could add the precondition to `supportsInterface` but I think the registering helper is the right place.)

Reply:
I just saw that there was a previous comment about this. I would still add this line just to be super clear that we're correctly implementing the spec.

Reply:
@frangio I thought the purpose of the magic value was to test the implementation as well; if the implementation, without the check, returns true for 0xffffffff, it shouldn't be trusted. And the default value for the key 0xffffffff in the supportedInterfaces map is false, so we theoretically shouldn't need a special check.

Like how in the ERC165MappingImplementation from https://github.com/ethereum/EIPs/blob/master/EIPS/eip-165.md#implementation they just say "don't set 0xffffffff to true".